diff -Nur fontforge-20100501.orig/fontforge/ffpython.h fontforge-20100501/fontforge/ffpython.h
--- fontforge-20100501.orig/fontforge/ffpython.h	2010-04-05 14:10:26.000000000 -0600
+++ fontforge-20100501/fontforge/ffpython.h	2011-02-16 20:28:37.608104668 -0700
@@ -62,7 +62,7 @@
 #define PyMODINIT_FUNC void
 #endif
 
-#if PY_MAJOR_VERSION < 2 || (PY_MAJOR_VERSION == 2 && PY_MINOR_VERSION <= 5)
+#if PY_MAJOR_VERSION < 2 || (PY_MAJOR_VERSION == 2 && PY_MINOR_VERSION <= 7)
 #define PyBytesObject PyStringObject
 #define PyBytes_Type PyString_Type
 
