--- fontforge-20061025/fontforge/tottf.c.fsSel	2006-12-09 21:36:24.000000000 +0330
+++ fontforge-20061025/fontforge/tottf.c	2006-12-09 21:41:45.000000000 +0330
@@ -3090,7 +3090,7 @@
     os2->fsSel = (at->head.macstyle&1?32:0)|(at->head.macstyle&2?1:0);
     if ( sf->fullname!=NULL && strstrmatch(sf->fullname,"outline")!=NULL )
 	os2->fsSel |= 8;
-    if ( os2->fsSel==0 ) os2->fsSel = 64;		/* Regular */
+    if ( os2->fsSel==0 && sf->pfminfo.weight == 400) os2->fsSel = 64;		/* Regular */
     if ( os2->version>=4 ) {
 	if ( strstrmatch(sf->fontname,"Obli")==0 ) {
 	    os2->fsSel &= ~1;		/* Turn off Italic */
