diff -Nur fontforge-20100501/fontforge/fvcomposit.c fontforge-20100501.orig/fontforge/fvcomposit.c
--- fontforge-20100501/fontforge/fvcomposit.c	2011-02-16 20:33:29.632880857 -0700
+++ fontforge-20100501.orig/fontforge/fvcomposit.c	2010-04-05 14:10:26.000000000 -0600
@@ -1337,7 +1337,7 @@
 	     unicodeenc==0x1fef || unicodeenc==0x1ffd || unicodeenc==0x1ffe))
 return( false );
 
-    if ( iszerowidth(unicodeenc) ||
+    if ((unicodeenc <= 0xffff && iszerowidth(unicodeenc)) ||
 	    (unicodeenc>=0x2000 && unicodeenc<=0x2015 ))
 return( !onlyaccents );
 
